fix(agw): Remove cyclic dependency between Logging and ITTI

lionelgo opened the pull request:
## Summary

Linked to issue #9732 

Replaced in lte/gateway/c/core/oai/lib/itti/*.c all calls to OAILOG_* by fprintf()

lionelgo committed:
Replace OAI_LOG by fprinf in lib/itti

Signed-off-by: lionelgo <29477918+lionelgo@users.noreply.github.com>

ssanadhya reviewed (changes requested):
Can you also remove the include for "log.h" from `lte/gateway/c/core/oai/lib/itti/intertask_interface.c`? 

Please also include a comment in ITTI_DEBUG messages that stdout is redirected to syslog when MME is run via systemd.

lionelgo committed:
remove the include for "log.h" from lte/gateway/c/core/oai/lib/itti/intertask_interface.c
include a comment in ITTI_DEBUG messages that stdout is redirected to syslog when MME is run via systemd

Signed-off-by: lionelgo <29477918+lionelgo@users.noreply.github.com>

lionelgo committed:
Missing liblfds include in ITTI

Signed-off-by: lionelgo <29477918+lionelgo@users.noreply.github.com>

lionelgo committed:
clang format on lte/gateway/c/core/oai/lib/itti/intertask_interface.c

Signed-off-by: lionelgo <29477918+lionelgo@users.noreply.github.com>

lionelgo commented:
Jenkins pipeline failed:
Bringing machine 'magma' up with 'libvirt' provider...
==> magma: You assigned a static IP ending in ".1" to this machine.
==> magma: This is very often used by the router and can cause the
==> magma: network to not work properly. If the network doesn't work
==> magma: properly, try changing this IP.
==> magma: Box 'magmacore/magma_dev' could not be found. Attempting to find and install...
    magma: Box Provider: libvirt
    magma: Box Version: 1.1.20210618
==> magma: Loading metadata for box 'magmacore/magma_dev'
    magma: URL: https://vagrantcloud.com/magmacore/magma_dev
The box you're attempting to add has no available version that
matches the constraints you requested. Please double-check your
settings. Also verify that if you specified version constraints,
that the provider you wish to use is available for these constraints.

Box: magmacore/magma_dev
Address: https://vagrantcloud.com/magmacore/magma_dev
Constraints: 1.1.20210618
Available versions: 1.1.20201124, 1.1.20201207, 1.1.20201216, 1.1.20201221, 1.1.20210120, 1.1.20210323, 1.1.20210325, 1.1.20210326, 1.1.20210601, 1.1.20210610, 1.1.20210611, 1.1.20210618, 1.1.20210716, 1.1.20210928

ssanadhya reviewed (approved):
LGTM

ssanadhya commented:
@lionelgo , could you please add a test plan too?

themarwhal commented on lte/gateway/c/core/oai/lib/itti/intertask_interface.c:
```diff
@@ -37,6 +37,7 @@
 #include <string.h>
 #include <malloc.h>
 #include <stdint.h>
+#include <liblfds710.h>
```
Just curious, which function is this dependency for?

themarwhal replied:
(@lionelgo)

lionelgo replied:
This is for this macro : LFDS710_MISC_MAKE_VALID_ON_CURRENT_LOGICAL_CORE_INITS_COMPLETED_BEFORE_NOW_ON_ANY_OTHER_LOGICAL_CORE. (But may be we don't need this lib anymore)

themarwhal replied:
Ah OK. So maybe we were pulling in the lfds header file somewhere up in the dependency before this change? Could you elaborate on why we might not need it anymore?

lionelgo replied:
We can still see some references to this lib in [lte/gateway/c/core/oai/common/shared_ts_log.h ](https://github.com/magma/magma/blob/58d67382cf011d82576ff5884c82c7a5e9dea3ab/lte/gateway/c/core/oai/common/shared_ts_log.h#L62)
